website_form_validations/controllers/main.py

```diff
@@ -0,0 +1,218 @@
+# -*- coding: utf-8 -*-
+from openerp import SUPERUSER_ID
+from openerp.http import request
+from openerp.addons.website_sale.controllers.main import website_sale
+
+
+class website_sale_inh(website_sale):
+
+    def checkout_values(self, data=None):
+        cr, uid, context, registry =\
+            request.cr, request.uid, request.context, request.registry
+        orm_partner = registry.get('res.partner')
+        orm_user = registry.get('res.users')
+        orm_country = registry.get('res.country')
+        state_orm = registry.get('res.country.state')
+
+        country_ids = orm_country.search(cr, SUPERUSER_ID, [], context=context)
+        countries = orm_country.browse(cr, SUPERUSER_ID, country_ids, context)
+        states_ids = state_orm.search(cr, SUPERUSER_ID, [], context=context)
+        states = state_orm.browse(cr, SUPERUSER_ID, states_ids, context)
+        partner = orm_user.browse(
+            cr,
+            SUPERUSER_ID,
+            request.uid,
+            context).partner_id
+
+        order = None
+
+        shipping_id = None
+        shipping_ids = []
+        checkout = {}
+        if not data:
+            if request.uid != request.website.user_id.id:
+                checkout.update(self.checkout_parse("billing", partner))
+                shipping_ids = orm_partner.search(
+                    cr, SUPERUSER_ID, [
+                        ("parent_id", "=", partner.id),
+                        ('type', "=", 'delivery')], context=context)
+            else:
+                order = request.website.sale_get_order(
+                    force_create=1,
+                    context=context)
+                if order.partner_id:
+                    domain = [("partner_id", "=", order.partner_id.id)]
+                    user_ids = request.registry['res.users'].search(
+                        cr,
+                        SUPERUSER_ID,
+                        domain,
+                        context=dict(
+                            context or {},
+                            active_test=False))
+                    if not user_ids or request.website.user_id.id not in user_ids:  # noqa
+                        checkout.update(
+                            self.checkout_parse(
+                                "billing",
+                                order.partner_id))
+        else:
+            checkout = self.checkout_parse('billing', data)
+            try:
+                shipping_id = int(data["shipping_id"])
+            except ValueError:
```

Comment:
To avoid use the try you can create a validation method and use the method is_digit

e.g.

def validate_shipping(ship):
    if ship[0] in ('-', '+'):
    	return ship[1:].isdigit()
    return ship.isdigit()